### ISSUE TITLE
Inverted activation

### ISSUE THREAD

**Issue by bgagl** [nistats/nistats#274]
Hi there, 

I am encountering an issue where I ran out of options.

The results from nistats are exactly inverted to my expectation in multiple investigations. Most prominent is the negative activation for a button press condition in relation to baseline (no other button press condition in all of the experiment), using the left hand, in the right motor cortex. Please see https://www.dropbox.com/s/m9foi714gkjuvxn/catch_trails_baselinecontrast.png?dl=0

I used the following code for contrast coding and plotting
```
zmap = second_level_model.compute_contrast(first_level_contrast='CATCH')
plotting.plot_glass_brain(zmap, colorbar=True, threshold=norm.isf(0.001),
                          title='CATCH (unc p<0.001)',
                          plot_abs=False, display_mode='lyrz')
```
the fitted models had the following form 

```
FirstLevelModel(drift_model='cosine', drift_order=1, fir_delays=[0],
        hrf_model='glover', mask=None, memory=Memory(cachedir=None),
        memory_level=1, min_onset=-24, minimize_memory=True, n_jobs=1,
        noise_model='ar1', period_cut=128, signal_scaling=True,
        slice_time_ref=0.0, smoothing_fwhm=8.0, standardize=False,
        subject_label='3IK02', t_r=2.25, target_affine=None,
        target_shape=None, verbose=0)
```
using the first_level_models_from_bids function. 

In addition, I used fMRIprep for preprocessing and Dcm2niix to convert my dicoms to niftis.

Any suggestions where this might originate from?
Thanks in advance.
Best wishes, 
Benjamin

**Comment by bthirion** [comment]
I'm a bit afraid of the FIR model here: it typically does not capture well the hrf, especially because you're only using a 1-lag model.
First, unless there is one good reason to do otherwise, I would use the `spm` hrf. 
The if you use a fir model, you probably want to use another, e.g. 4s. 
Does it help ?

**Comment by bgagl** [comment]
Thanks for the response 

I ran also the following model 

```
FirstLevelModel(drift_model='cosine', drift_order=1, fir_delays=[0],
        hrf_model='spm + derivative + dispersion', mask=None,
        memory=Memory(cachedir=None), memory_level=1, min_onset=-24,
        minimize_memory=True, n_jobs=1, noise_model='ar1', period_cut=128,
        signal_scaling=True, slice_time_ref=0.0, smoothing_fwhm=6.0,
        standardize=False, subject_label='3IK02', t_r=2.25,
        target_affine=None, target_shape=None, verbose=0)
```

No change in the result. Or did I get something wrong here?

**Comment by bthirion** [comment]
Have you checked that the contrasts, especially those of the second-level analysis are well specified ?
You probably want to  use `plot_contrast_matrix`.
You can also look at the individual maps to see if the inconsistency is at the first or second level.

**Comment by bgagl** [comment]
It looks like the problem is there on the first level already (see participant 1-4 here https://www.dropbox.com/sh/b9j6kl5s9pa681x/AADa9qN0wdjCxJ9N2QatN2d_a?dl=0). In addition, there is a contrast matrix in the same folder. Looks ok to me.

here is the related code.
```

        model.fit(imgs, events, confounds)
        plot_design_matrix(model.design_matrices_[0])
        zmap = model.compute_contrast(['CATCH'])
        plotting.plot_glass_brain(zmap, colorbar=True, threshold=norm.isf(0.001),
                          title='CATCH (unc p<'+str(p)+')',
                          plot_abs=False, display_mode='lyrz') 
        plt.show()
```

Thanks, I hope this can give you a clue.

**Comment by bthirion** [comment]
* The CSF, WM and GM regressors seem to be flat. Are they 0 ? If yes, this is an issue: the GLM may not work well. 
* Also, I would remove the dispersion derivative.
* Is there any contrast more meaningful than 'CATCH' alone ? Why not 'CATCH-CS' or 'CATH-PW' (sorry if this is meaningful) ? Such contrasts may better outline effects of interest.
* Finally, I am wondering whether your data were mean-centered prior to GLM fitting. Is it the case ? (to know that, compute the mean bold image of the series, and look at the values of this mean image.)

**Comment by bgagl** [comment]
Quick update, here is the complete notebook of the nistats analysis: https://www.dropbox.com/s/89yptesdhj4yti0/first_second_level_ext_LCM-W_PW_CS.html?dl=0

**Comment by bgagl** [comment]
Quick update, here is the complete notebook of the nistats analysis now EXCLUDING the CSF, WM and GM regressors ... does not look good: https://www.dropbox.com/s/5fm4yhhvmcjhzfb/first_second_level_ext_LCM-W_PW_CS.html?dl=0

**Comment by bthirion** [comment]
OK, will look at it today hopefully.
B

**Comment by bthirion** [comment]
Sorry, I've been distracted by a bunch of things. Here are a few comments:
* I find the design matrix for subject "AFN02" fishy: there is no event occurring at the beginning... Are you sure that the onset times a re right ?
* you have to upgrade nistats (see install page),as you are not using the latest version.
* there is a problem with subjects OVS14 ADA07 ALA20 ASA05 EBN14 GJA23 RVA30 SKK20 YBA08 AKA01 and  BWC24
* In general, the values after the first-level analysis that are output are pretty packed around zero. This is strange.  Are their some artefact in the data that would decrease the amount of explained variance ? 
*  Even under chance, you shoud; obtain more extreme values. Have you checked the brain mask that you are using ?
* if possible I would use 'CATCH-something else' instead of  'CATCH' depending on the conditions you have. It may remove some weird individual effect
* At least for the bad subjects, I would try to fit separately run1 and run2, and see if one of them is particularly problematic. It might be worth rejecting data if you find some kind of corruption.

*  the second-level results are a mess, but this probably comes from wrong first level analyses
HTH

**Comment by bthirion** [comment]
Sorry, I've been distracted by a bunch of things. Here are a few comments:
* I find the design matrix for subject "AFN02" fishy: there is no event occurring at the beginning... Are you sure that the onset times a re right ?
* you have to upgrade nistats (see install page),as you are not using the latest version.
* there is a problem with subjects OVS14 ADA07 ALA20 ASA05 EBN14 GJA23 RVA30 SKK20 YBA08 AKA01 and  BWC24
* In general, the values after the first-level analysis that are output are pretty packed around zero. This is strange.  Are their some artefact in the data that would decrease the amount of explained variance ? 
*  Even under chance, you shoud; obtain more extreme values. Have you checked the brain mask that you are using ?
* if possible I would use 'CATCH-something else' instead of  'CATCH' depending on the conditions you have. It may remove some weird individual effect
* At least for the bad subjects, I would try to fit separately run1 and run2, and see if one of them is particularly problematic. It might be worth rejecting data if you find some kind of corruption.

*  the second-level results are a mess, but this probably comes from wrong first level analyses
HTH

**Comment by bgagl** [comment]
Hi there, thanks for the last comment, this lead to identification of an error prior to stats. Fist checks and analyses look good. 

Once more thanks for the great help ! I am becoming a big fan of nistats

**Comment by bthirion** [comment]
Thx ! 
Please close the issue if you think it's fine.
Best,
